### DISCUSSION TITLE
Span Normalization for OPTIONS http verb

### DISCUSSION THREAD

**Discussion by venkatramanchandirasekaran** [open-telemetry/opentelemetry-dotnet#3785] [Q&A]
I am using Open Telemetry dotnet library(1.1.0) to send traces to Splunk APM product and the framework does a good job at normalizing the Spans. The problem i am seeing is that for the OPTIONS HTTP verb / if there are HTTP errors, there is no Span normalization. When will that be supported ?

1) **No Span Normalization on a HTTP error (ex: 404)**
![Span-Normalization-404](https://user-images.githubusercontent.com/95233657/196550425-a3f8fac4-f488-432a-b790-0d95d1159677.png)

2) **No Span Normalization on an OPTIONS HTTP verb**

3) Example - Span Normalization - Working on an GET
![Span-Normalization-Working](https://user-images.githubusercontent.com/95233657/196550915-a75e3498-dc14-4029-9bb8-5eea0d42f3f4.png)

Here is the open telemetry version i am using
![open-telemetry-versions](https://user-images.githubusercontent.com/95233657/196549991-c5133810-da73-4b49-92b5-d3d253dc5ea5.png)

**Comment by venkatramanchandirasekaran** [comment]
Are there any updates to the question i have ?

**Reply by pjanotti** [reply]
Hi @venkatramanchandirasekaran I did a quick test using a test app and didn't get a repro using the current sources and or when using `PackageReference Include="OpenTelemetry.Instrumentation.AspNetCore" Version="1.0.0-rc9.11"`. Here is what I get:

```terminal
Activity.TraceId:            13f06cb0bd8f325c7573908a1651c711
Activity.SpanId:             6ec482a92f956433
Activity.TraceFlags:         Recorded
Activity.ActivitySourceName: Microsoft.AspNetCore
Activity.DisplayName:        api/Values/{id}
Activity.Kind:               Server
Activity.StartTime:          2023-01-24T00:05:42.6824564Z
Activity.Duration:           00:00:00.0003915
Activity.Tags:
    net.host.name: localhost
    net.host.port: 50511
    http.method: OPTIONS
    http.scheme: https
    http.target: /api/values/-2
    http.url: https://localhost:50511/api/values/-2
    http.flavor: 1.1
    http.user_agent: Mozilla/5.0 (Windows NT 10.0; Microsoft Windows 10.0.19044; en-US) PowerShell/7.3.1
    http.route: api/Values/{id}
    http.status_code: 404
Resource associated with Activity:
    service.name: unknown_service:TestApp.AspNetCore

Activity.TraceId:            672162c60706b1f9b977e3e86ab7b844
Activity.SpanId:             31f1300040a0dbe7
Activity.TraceFlags:         Recorded
Activity.ActivitySourceName: Microsoft.AspNetCore
Activity.DisplayName:        api/Values/{id}
Activity.Kind:               Server
Activity.StartTime:          2023-01-24T00:09:38.5808554Z
Activity.Duration:           00:00:00.0010118
Activity.Tags:
    net.host.name: localhost
    net.host.port: 50511
    http.method: GET
    http.scheme: https
    http.target: /api/values/-2
    http.url: https://localhost:50511/api/values/-2
    http.flavor: 1.1
    http.user_agent: Mozilla/5.0 (Windows NT 10.0; Microsoft Windows 10.0.19044; en-US) PowerShell/7.3.1
    http.route: api/Values/{id}
    http.status_code: 404
Resource associated with Activity:
    service.name: unknown_service:TestApp.AspNetCore
```

**Reply by venkatramanchandirasekaran** [reply]
Thanks a lot. I will update to latest OTEL library and test it.
